aspnetcore/blazor/fundamentals/render-modes.md

```diff
@@ -0,0 +1,295 @@
+---
+title: ASP.NET Core Blazor render modes
+author: guardrex
+description: Learn about Blazor render modes and how to apply them in Blazor Web Apps.
+monikerRange: '>= aspnetcore-8.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 09/08/2023
+uid: blazor/fundamentals/render-modes
+---
+# ASP.NET Core Blazor render modes
+
+This article explains control of Razor component rendering in Blazor Web Apps, either at compile time or runtime.
+
+## Render modes
```

Comment:
We should recommend that component authors implement components to be render mode agnostic. It should typically be possible to use a component with any render mode. The app developer generally decides what render mode to use for a given component using the `@rendermode` directive on the component instance.

The main exception to this is when defining a page. To specify the render mode for an entire page you need to use the `@attribute` directive and the corresponding render mode attribute. But since pages are typically defined by app developers, not library authors, it's still the app developer that is in control of the render mode.

This doc currently focuses heavily on the render mode attributes. But given the above guidance it would be more appropriate to focus on using the `@rendermode` directive instead of tying a component's definition to a specific render mode.